tools/options.c

```diff
@@ -345,7 +345,9 @@ static void nn_append_string (struct nn_parse_context *ctx,
     if (!lst->items) {
         nn_memory_error (ctx);
     }
-    lst->items[lst->num-1] = str;
```

Comment:
This is a logic change.  It shouldn't matter because nn_memory_error (ctx) should probably abort the program.  Maybe that needs a NORETURN (or whatever) statement?  I actually don't like adding the extra indentation here, it makes the flow harder to grok.